### DISCUSSION TITLE
proxytunnel's default cacert directory is always empty on macOS

### DISCUSSION THREAD

**Discussion by deFractal** [Homebrew/discussions#1933] [Writing Formulae/Casks]
Currently, `proxytunnel` looks by default for a CA certificate file in /etc/ssl/certs, but on current versions of macOS, that directory is always empty. Is there a practical means to alter the formula for `proxytunnel` on macOS to default instead to the cacert.pem file corresponding to the OpenSSL library the `proxytunnel` binary is linked against? 

So, for example, my scripts which use `proxytunnel` currently call it with `--cacert="$(/usr/local/opt/openssl/bin/openssl version -d | cut -d'"' -f2)/cacert.pem"` (though perhaps there's a more elegant way to write this). Currently, that substitution maps to `--cacert=/usr/local/etc/openssl@1.1/cacert.pem`. Without that, `proxytunnel` always fails to validate any certificates. If this were addressed, the `--cacert` argument could be omitted in this case, because `proxytunnel` would look there by default. 

I'm not sure yet where it would make the most sense to create an issue for this. Recommendations welcome.

**Comment by carlocab** [comment]
I'd look at their `Makefile` to see if there's a way to configure the default CA cert location at build time, so we can just do this in the formula.

**Reply by deFractal** [reply]
Before I try to patch their `Makefile`, could you offer any feedback on what I'm trying to do in [this patch to their ptstream.c](https://github.com/deFractal/proxytunnel/commit/e448313d686bd4ef3b99778ea88f666a3c82e040)?

It's been quite a while since I've done anything serious with C, so I'm not certain if I'm on the right track there.

**Reply by deFractal** [reply]
I suppose it would be better to let this be more general, instead of substituting one hard-coded path for another. What are your thoughts on [this change set](https://github.com/proxytunnel/proxytunnel/compare/master...deFractal:homebrew-mac-makefile?expand=1), @carlocab?

**Reply by carlocab** [reply]
The latest changes to `ptstream.c` make sense to me, though I think we don't want to assume a `/usr/local` prefix in their `Makefile`. Though, given that their `Makefile` seems to be designed to be edited manually by commenting certain things in and out, I suppose your additions are fine as an example for users who want to modify it.

**Reply by deFractal** [reply]
Yes, I'd originally excluded the `Makefile` while testing, and I'll excluded it also when I file a PR, but I thought I'd share what configuration I'd used to test the changes to `ptstream.c`. 

I've verified this compiles and runs effectively for me, as a drop-in replacement for `proxytunnel` as ships with Homebrew on macOS Catalina. 

<details>
<summary>For example, I can use the resulting binary for an SSH/HTTPS tunnel, …</summary> after running `make` with only the changes linked above, by replacing:

```
ProxyCommand proxytunnel --quiet --encrypt-proxy --cacert="$(/usr/local/opt/openssl/bin/openssl version -d | cut -d'"' -f2)/cacert.pem" --proxy=some-server.invalid:443 --dest=127.0.0.1:22
```
with

```
ProxyCommand ~/Documents/GitHub/proxytunnel/proxytunnel --quiet --encrypt-proxy --proxy=some-server.invalid:443 --dest=127.0.0.1:22
```
in `.ssh/config`.
</details>

Would you suggest a PR to proxytunnel/proxytunnel first, then to Homebrew/homebrew-core? (Or…?)

**Reply by deFractal** [reply]
Alright, @carlocab, as of [this commit](https://github.com/proxytunnel/proxytunnel/commit/e2a214d942c46268cf427341eacb296428079026), it's configurable in their `Makefile` now. All we have to do to build the new upstream on macOS (besides having the dependencies in place) is to uncomment the line for "DARWIN" and the four lines for "DARWIN, continued" in the new `Makefile`. 

Now I just have [some reading to do](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md) before I can begin to draft a PR for Homebrew/homebrew-core. Thank you for pointing me in the right direction.

**Reply by carlocab** [reply]
Thanks for working on this! Uncommenting those lines won't quite work for us, as I mentioned above, since we don't want to assume in the formula that Homebrew is installed in `/usr/local`.

Your changes will probably also have to make it into a stable release of proxytunnel before we can use them. Alternatively, we can possibly apply a patch in the formula using the commit from your PR.

**Reply by deFractal** [reply]
I think I've got how to edit the `Makefile` to use whichever prefix Homebrew's installed in. 

Perhaps [something like this](https://github.com/deFractal/proxytunnel/blob/homebrew-mac-makefile/Makefile)? 

IIUC, to avoid assuming a `/usr/local` prefix, the version I just pushed of the linked `Makefile` requires one other change to the formula besides the URL and hashes, particularly to the second line of the formula's `def install` block:
```ruby
    system "make", "prefix=#{prefix}"
```

**Reply by carlocab** [reply]
That doesn't quite work, since proxytunnel's install `prefix` is different from the prefix we want to use for OpenSSL. Don't worry about it -- we can always pass the appropriate `OPTFLAGS`, etc directly to `make`.

**Reply by deFractal** [reply]
Makes sense. Feel free to modify my first attempt there, if you'd like, as it'll be a while before I can get back to it. I'd only tested that it compiles correctly, and I can see now it'd need to be tweaked for `make install`.
